daemon/cdi.go

```diff
@@ -0,0 +1,38 @@
+package daemon
+
+import (
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	"github.com/docker/docker/errdefs"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var cdiRegistry cdi.Registry
+
+// initializeCdi initializes the CDI registry and registers the CDI device driver.
+func initializeCdi(d *Daemon) error {
+	features := d.Features()
+	if features == nil {
+		return nil
+	}
+
+	if cdiEnabled, ok := (*features)["cdi"]; !ok || !cdiEnabled {
+		return nil
+	}
+
+	if !d.HasExperimental() {
+		return errdefs.InvalidParameter(errors.New("the cdi featuer is only supported in experimental mode"))
```

Comment:
Noticed while looking at the test: Do we need both the experimental flag _and_ the default-off feature flag gating this? Is the intent to keep the feature flag around and default-off "forever", even after the Experimental check is removed?

My _personal_ inclination would be to drop the "Experimental" check now, and just use the feature flag, if there's concerns about people using this feature by accident, since it's more fine-grained. It also means that if-and-when we want this feature default-on (as I'd hope we do), then users can pre-override that with an explicit `false` for the feature.

Conversely, having the experimental flag and no feature flag would also make sense, if we assume no one's going to want to actively disable CDI support. I don't know if CDI has security risk factors or otherwise that might make this a thing Docker administrators actively consider.

Reply:
@corhere do you have objections to only using the `experimental` flag? From my point of view, I think the goal would be to have this as default on at some point but I'm not sure whether users would want finer-grained control.

Reply:
Note that I have reworked this so that the CDI registry (implemented as a singleton) is only instantiated if a CDI device is actually requested and as such the impact should be minimal for users where no such devices are being used.

For now I will simplify this to only use the `experimental` flag and we can consider adding an additional feature gate if desired.

Reply:
Gating behind only the `experimental` flag is fine for now.

Reply:
This has been updated to only use the `experimental` flag.